lib/file-store.js

```diff
@@ -17,24 +18,22 @@ const sql = sqlLoader.loadSqlEquiv(__filename);
  * @param {number|null} instance_question_id - The instance question for the file.
  * @param {number} user_id - The current user performing the update.
  * @param {number} authn_user_id - The current authenticated user.
+ * @param {string} storage_type - AWS 's3' or 'filesystem' storage options.
  * @return {number} The file_id of the newly created file.
  */
-module.exports.upload = async (display_filename, contents, type, assessment_instance_id, instance_question_id, user_id, authn_user_id) => {
-    debug('upload()');
-    // Make a filename to store the file. We use a UUIDv4 as the filename,
-    // and put it in two levels of directories corresponding to the first-3
-    // and second-3 characters of the filename.
+
+module.exports.storageTypes = Object.freeze({S3: 'S3', FileSystem: 'FileSystem'});
+
+module.exports.upload = async (display_filename, buffer, type, assessment_instance_id, instance_question_id, user_id, authn_user_id, storage_type) => {
+    // default S3
+    if (!storage_type) storage_type = this.storageTypes.S3;
```

Comment:
Could we add a config variable for this default? Maybe `config.fileStoreStorageDefault = 'S3'`?

Reply:
Added here: dfc65ae9b6d5bf843d6aa0d093620ff675836e7a.